vsock/README: update qemu example

stefano-garzarella opened the pull request:
As pointed out in https://gitlab.com/qemu-project/qemu/-/issues/1337 `-numa` is not supported by all machine types, so let's update the example using the `memory-backend` machine property.

Since we are modifying the example, let's clean up a little bit, for example by removing `-enable-kvm` and using machine properties to enable acceleration.

Signed-off-by: Stefano Garzarella <sgarzare@redhat.com>

stsquad commented on crates/vsock/README.md:
```diff
@@ -46,11 +46,11 @@ vhost-user-vsock --guest-cid=4 --uds-path=/tmp/vm4.vsock --socket=/tmp/vhost4.so
-  -device vhost-user-vsock-pci,chardev=char0 \
-  -object memory-backend-file,share=on,id=mem,size="512M",mem-path="/dev/hugepages" \
-  -numa node,memdev=mem -mem-prealloc
+qemu-system-x86_64 -drive file=/path/to/disk.qcow2 -smp 2 \
```
It might be easier to separate all the non vsock specific bits in the example to avoid confusion, e.g.:

```
qemu-system-x86_64 \
  ... normal machine options ... \
  -object memory-backend-file,share=on,id=mem0,size=512M,mem-path="/dev/hugepages" \ # size == -m size
  -chardev socket,id=char0,reconnect=0,path=/tmp/vhost4.socket \
  -device vhost-user-vsock-pci,chardev=char0
```

possibly follow with a fully working example?

stefano-garzarella replied:
Good point, I'll update it!

stefano-garzarella commented:
v2:
- provided only the QEMU's vhost-user-vsock required options in the "Usage" section
- added "Working example" section
- removed spurious rust section at the bottom of the file

stsquad commented:
I'm assuming the buildkite failures are a temp hiccup. Is there anyway to re-trigger the tests?

stefano-garzarella commented:
> I'm assuming the buildkite failures are a temp hiccup. Is there anyway to re-trigger the tests?

I think so.
@andreeaflorescu how can we re-trigger it? Force push?

andreeaflorescu commented:
> > I'm assuming the buildkite failures are a temp hiccup. Is there anyway to re-trigger the tests?
> 
> I think so. @andreeaflorescu how can we re-trigger it? Force push?

Not necessarily, there are 3 ways to retrigger:
- if you are an admin of the github repo, you can go to the Settings -> Webhooks -> Edit -> Recent Deliveries -> Find the corresponding event (i.e.  pull_request.synchronize) and then click on Redeliver; this will retrigger all tests
- force push, the only option directly available for non-admins
- trigger just the steps that failed from Buildkite, you need to be a Buildkite admin for that. I can add more people to Buildkite as well, let me know if you want me to add you.

andreeaflorescu commented:
The tests will fail for this PR because of the advisory (https://buildkite.com/rust-vmm/vhost-device-ci/builds/985#0184c420-e373-4758-ba2e-7f89e7aefb3a). More details here: https://github.com/rust-vmm/rust-vmm-ci/issues/114

You can also "fix" this locally using the instructions here: https://github.com/rust-vmm/rust-vmm-ci/issues/114#issuecomment-1330888057

andreeaflorescu commented:
I retriggered the other steps that failed, but it looks like they don't pass, they're running on the mshv hosts, so I think that those behave differently from the EC2 ones that fetch the containers and then seem to fit the timeout.

vireshk commented:
> * trigger just the steps that failed from Buildkite, you need to be a Buildkite admin for that. I can add more people to Buildkite as well, let me know if you want me to add you.

Please add me, if possible. Thanks.

stefano-garzarella commented:
v3:
- rebased on main with f4a47df5a14ab5204174eac6264437a1456fac6c merged

vireshk commented:
> v3:
> 
> * rebased on main with [f4a47df](https://github.com/rust-vmm/vhost-device/commit/f4a47df5a14ab5204174eac6264437a1456fac6c) merged

The audit test did pass, but the timeout issue is not letting us go through :(

andreeaflorescu commented:
I will be working on a fix for that today, or at least provide a workaround.
